libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -145,9 +162,32 @@ void AC_PrecLand::init()
     }
 }
 
+// create inertial buffer
+void AC_PrecLand::check_inertial_buffer()
+{
+    // if lag parameter is not within bounds, return without doing anything
+    if (_lag_ms <= 0 || _lag_ms > 250) {
+        return;
+    }
+
+    // calculate inertial buffer size from delay length (loop runs at 400hz)
+    uint16_t inertial_buffer_size = (uint16_t)roundf(_lag_ms / (1000/400.0f));
```

Comment:
Doesn't really matter, but this could be reordered to avoid a division and done without floats...

Also, with a limit of 250ms on _lag_ms this comes out to 100, within a uint8_t

Reply:
MichaelDB suggests LAG should be in seconds, I'm changing this buffer size calculation to be:

uint16_t inertial_buffer_size = MAX((uint16_t)roundf(_lag * AP::scheduler().get_loop_rate_hz()), 1);